### ISSUE TITLE
Scaladoc: Too many deprecated methods

### ISSUE THREAD

**Issue by ashawley** [scala/bug#11393]
There are a bunch of methods listed as deprecated for [List](https://www.scala-lang.org/api/2.13.0-M5/scala/collection/immutable/List.html) in 2.13.  It doesn't appear they're actually deprecated.  The method names are in the color green.  Is it something to do with deprecations in [IterableOnceExtensionMethods](https://www.scala-lang.org/api/2.13.0-M5/scala/collection/IterableOnceExtensionMethods.html).

```
collectFirst
copyToBuffer
count
exists
filter
find
flatMap
foldLeft
foldRight
forall
foreach
isEmpty
map
[... 31 others]
```

These are implicit methods for `List` but they are not deprecated, and they should be shadowed in Scaladoc.  This unnecessarily clutters up the Scaladocs but also dilutes the list of deprecated members.  Granted, most people probably learn about their deprecations from interactive compiler warnings and not the api docs.

**Comment by szeiger** [comment]
In some cases like `copyToBuffer` the deprecated methods are legitimate. Others like `collectFirst` and `count` are listed twice, both as regular (non-deprecated) methods and as deprecated extension methods. This is another instance of https://github.com/scala/bug/issues/6110.

**Comment by Jasper-M** [comment]
The scaladoc also says they are shadowed, and mentions a way to call them anyway (`(list: IterableOnceExtensionMethods[A]).count(f)`).

**Comment by szeiger** [comment]
Yeah, it looks like this is not an actual bug in scaladoc (anymore). I didn't rerun #6110 but I assume it's in a similar state these days. scaladoc already gets the shadowing right, the real question is what to do with that information. Is it safe to assume that an instance method or extension methods "overrides" a shadowed, less specific extension method? Unlike overridden methods, these shadowed methods can still be called through a simple type ascription. Sorting the methods into deprecated and non-deprecated first and only mentioning shadowing individually makes it hard to convey this information.
